RelationBehavior.php

```diff
@@ -162,7 +162,16 @@ protected function loadData()
             $data['newModels'] = [];
             $class = $activeQuery->modelClass;
 
-            if (!ArrayHelper::isAssociative($activeQuery->on) && !empty($activeQuery->on)) {
+            if (
+                (!ArrayHelper::isAssociative($activeQuery->on) && !empty($activeQuery->on)) ||
+                (
+                    $activeQuery->multiple &&
+                    !empty($activeQuery->via) &&
+                    is_object($activeQuery->via[1]) &&
+                    !ArrayHelper::isAssociative($activeQuery->via[1]->on) &&
+                    !empty($activeQuery->via[1]->on)
```

Comment:
wrong check - it's will throw exception only if both "on" condition will incorrect. Should use "OR" logic instead. And why used is_object check instead ArrayHelper::isAssociative ?